Automattic/wp-calypso
# The selected plans state on the drop downs should not `reset` when there is a parent  state refresh

- Fixes Automattic/martech#2554

## Proposed Changes

The ComparisonGrid component now includes a "Compact Display" configuration tailored for smaller screens, streamlining plan feature comparisons. End-users can easily customize the order of plans to suit their preferences within the grid. To maintain order consistency, especially during unmounts triggered by plan interval changes, this state is now efficiently managed within the WpcomPlansUI store.

When a change in `intervalType` propagates down to the ComparisonGrid as a prop, the implemented solution in this PR detects this prop change. It then seamlessly transforms the current plan order into a set that aligns with the currently selected plan interval. This transformation is designed to avoid any glitches or flashes during the transition.

https://github.com/Automattic/wp-calypso/assets/3422709/0f69dabf-e91a-48de-a5d0-3c3d84977eb5

## Testing Instructions



* On the comparison grid, on smaller resolutions, make sure the selected dropdown order does not change on interval change

## Pre-merge Checklist
